smac/optimizer/acquisition.py

```diff
@@ -40,21 +40,21 @@ def __init__(self, model: AbstractEPM):
         self.logger = PickableLoggerAdapter(self.__module__ + "." + self.__class__.__name__)
 
     def update(self, **kwargs):
-        """Update the acquisition functions values.
+        """Update the acquisition function attributes required for calculation.
 
-        This method will be called if the model is updated. E.g.
-        entropy search uses it to update its approximation of P(x=x_min),
-        EI uses it to update the current fmin.
+        This method will be called after fitting the model, but before maximizing the acquisition
+        function. As an examples, EI uses it to update the current fmin.
 
-        The default implementation takes all keyword arguments and sets the
-        respective attributes for the acquisition function object.
+        The default implementation only updates the attributes of the acqusition function which
+        are already present.
 
         Parameters
         ----------
         kwargs
         """
         for key in kwargs:
-            setattr(self, key, kwargs[key])
+            if hasattr(self, key):
```

Comment:
Why shouldn't you throw an error if one tries to update an attribute that doesn't exist?

Reply:
The idea is that the SMBO class passes a large amount of values to the acquisition function, and only the ones which are needed by the acquisition function are stored. This way, the acquisition function does not need to know what can possibly be passed, and the smbo class does not need to know what arguments are required by the acquisition function.

Reply:
Yes, I just would like to avoid that users thinks they have set an attribute which is not set

Reply:
Okay, I just added such a check.